internal/project/core_test.go

```diff
@@ -53,3 +58,177 @@ func TestCore_ExistsWithID(t *testing.T) {
+	switch num {
+	case 1:
+		testProject = project1
+	case 2:
+		testProject = project2
+	case 3:
+		testProject = project3
+	}
+
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		want    *Model
+		wantErr bool
+	}{
+		{
+			name: "Get Project Successfully.",
+			fields: fields{
+				repo: mockRepo,
+			},
+			args: args{
+				ctx:       ctx,
+				projectID: testProject.ProjectID,
+			},
+			want:    testProject,
+			wantErr: false,
+		},
+		{
+			name: "Throw error because of Invalid Project ID.",
+			fields: fields{
+				repo: mockRepo,
+			},
+			args: args{
+				ctx:       ctx,
+				projectID: "",
+			},
+			want:    nil,
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &Core{
+				repo: tt.fields.repo,
+			}
+			var err2 error = nil
+			if len(tt.args.projectID) == 0 {
+				err2 = fmt.Errorf("Invalid Project ID!")
+			}
+			mockRepo.EXPECT().Get(gomock.Any(), gomock.Any(), &Model{}).Do(func(arg1 context.Context, arg2 string, mod *Model) {
+				if err2 == nil {
+					id := strings.Split(tt.args.projectID, "-")
+					serial := id[1]
+
+					mod.ProjectID = "testID" + "-" + serial
+					mod.Name = "test" + "-" + serial
+					mod.Labels = map[string]string{"label": "value"}
+					mod.SetVersion(serial)
+				}
+			}).Return(err2)
+			got, err := c.Get(tt.args.ctx, tt.args.projectID)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Core.Get() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("Core.Get() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestCore_DeleteProject(t *testing.T) {
+	type fields struct {
+		repo IRepo
+	}
+	type args struct {
+		ctx context.Context
+		m   *Model
+	}
+	ctrl := gomock.NewController(t)
+	mockRepo := mocks.NewMockIRepo(ctrl)
+	project := getDummyProjectModel()
+	ctx := context.Background()
+
+	dummyModel := &Model{
+		Name:      "test",
+		ProjectID: "",
+		Labels:    map[string]string{"label": "value"},
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Delete existing project with No errors.",
+			fields: fields{
+				repo: mockRepo,
+			},
+			args: args{
+				ctx: ctx,
+				m:   project,
+			},
+			wantErr: false,
+		},
+		{
+			name: "Get error as Project doesn't exist.",
+			fields: fields{
+				repo: mockRepo,
+			},
+			args: args{
+				ctx: ctx,
+				m:   dummyModel,
+			},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &Core{
+				repo: tt.fields.repo,
+			}
+			var err2 error = nil
+			if len(tt.args.m.ProjectID) == 0 {
+				err2 = fmt.Errorf("Invalid Project ID!")
+				mockRepo.EXPECT().Exists(gomock.Any(), tt.args.m.Key()).Return(false, err2)
+			} else {
+				mockRepo.EXPECT().Exists(gomock.Any(), tt.args.m.Key()).Return(true, err2)
+				mockRepo.EXPECT().Delete(gomock.Any(), tt.args.m).Return(err2)
```

Comment:
This should return `nil` and not err2. err2 is contextual to `Exists`

Reply:
But if I have initialized err2 = nil irrespective, then shouldn't this work.